api/http_lifecycle.go

```diff
@@ -21,28 +21,23 @@ var (
 	ErrNoOperationGiven = errors.New("lfs/api: no operation provided in schema")
 )
 
-// EndpointSource is an interface which encapsulates the behavior of returning
-// `config.Endpoint`s based on a particular operation.
-type EndpointSource interface {
```

Comment:
I deleted this because it's a subset of `Configuration`, and in fact this code needs the whole of that structure so the abstraction wasn't very useful and was hiding that the code internally was accessing the global config.